<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Fundamentals (Good) Exercises</span></div>

# Exercise 1 (f-strings to the rescue) 

In [11]:
user_input = input('What is your name?')
print('How you doin ', user_input,"!", sep='')

What is your name? MING RONG


How you doin MING RONG!


In [16]:
user_name = input('What is your name?')
user_age = input ('What is your age?')
print('My name is ', user_name, ' and I am ', user_age, ' years old.', sep='')

What is your name? MING RONG
What is your age? 22


My name is MING RONG and I am 22 years old.


In [33]:
pi = 3.141592
def hehe(x):
    print(f'{'The value of pi to'} {x} {'decimal places is: '}{pi:.{x}f}')

In [36]:
hehe(2)

The value of pi to 2 decimal places is: 3.14


In [37]:
hehe(3)

The value of pi to 3 decimal places is: 3.142


In [38]:
hehe(4)

The value of pi to 4 decimal places is: 3.1416


The value of pi to 2 decimal places is: 3.14 \
The value of pi to 3 decimal places is: 3.142 \
The value of pi to 4 decimal places is: 3.1416 

In [71]:
print(f'{'Apple ':<10}{'$0.99':>5}{'\n'}{'Banana ':<10}{'$0.59':>5}{'\n'}{'Orange ':<10}{'$1.29':>5}{'\n'}')


Apple     $0.99
Banana    $0.59
Orange    $1.29



# Exercise 2 (What is your grade?)

## WRONG

In [28]:
score_input = int(input("Enter the student's score: "))
si = score_input 

def grade(si):
    if si >= 70:
        print('A')
    elif si >= 50:
        print('B')
    elif si >= 35:
        print('C')
    elif si >= 0:
        print('Fail')  #'grade(si)' function will print the grade before the next print function

print(f"{"The student's letter grade is: "}{grade(si)}")

Enter the student's score:  99


A
The student's letter grade is: None


## FINAL ANS

In [2]:
score_input = int(input("Enter the student's score: "))
si = score_input 

def grade(si):
    if si >= 70:
        x='A'
    elif si >= 50:
        x='B'
    elif si >= 35:
        x='C'
    elif si >= 0:
        x='Fail'  #'grade(si)' function will print the grade before the next print function
    return x

print(f"{"The student's letter grade is: "}{grade(si)}")

Enter the student's score:  87


The student's letter grade is: A


# Exercise 3 (Debugging Code)

In [15]:
def factorial(n):
    if n < 0:
        return "Invalid input"
    elif n == 0:
        return 1
    else:
        result = 1
        for number in range(1, n+1):
            result *= number
    return result


print(factorial(5))   # Testing, expected output: 120
print(factorial(-1))  # Testing, expected output: Invalid input

120
Invalid input
